File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.java
class Solution {
    public int missingMultiple(int[] nums, int k) {
        long[] x = {0L, 0L};
        for (int n : nums)
            if (n % k == 0) {
                int i = n / k - 1;
                x[i >> 6] |= 1L << (i & 63);
            }
            
        int z = x[0] == -1L ? 1 : 0;
        return (z * 64 + Long.numberOfTrailingZeros(++x[z] & -x[z]) + 1) * k;
    }
}

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.cpp
class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        uint64_t x[2] = {0, 0};
        for (auto& n : nums)
            if (n % k == 0) {
                int i = n / k - 1;
                x[i >> 6] |= 1ULL << (i & 63);
            }

        int z = x[0] == -1ULL;
        return (z * 64 + bit_width(++x[z] & -x[z])) * k;
    }
};